### DISCUSSION TITLE
Keep text selection after dragging out of node

### DISCUSSION THREAD

**Discussion by TobiasKrok** [xyflow/xyflow#2942] [Q&A]
Hi!

I am making a node with a text editor  (Quilljs) and I am having some issues with text selection. If I drag and select text inside the editor and my mouse leaves the node, the text becomes deselected.

Please take a look at this quick example (using html text inputs, not Quill): https://codesandbox.io/s/immutable-framework-p23h6w?file=/src/App.js

I created a custom test node named TextNode

If you write something in the text area or text input and drag select so your mouse leaves the node, it becomes unselected. If you drag a little bit further while holding left click, it selects in again. How can I keep the selection consistent and keep the selection when you drag outside of the node (and let go of left click)?

Thank you!

How can I keep the selection when you leave the node and

**Comment by Josso7** [comment]
Hello,

I discovered identical behavior on an input field. I would select and drag text in the input and once my mouse left the containing wrapper div the selected text would get partially unselected back to the center of my input. It would then reselect text as I continued to move the cursor in the same direction. I was able to narrow it down to a single css property causing this strange behavior. I was using `transform: translate(-50%, -50%);` on the inputs wrapper div. Once I removed this, my input would behave completely normally and moving the cursor outside the div no longer deselected text. 

I don't use react-flow but your post was the only issue I could find similar to mine. Apparently modifying an element with the transform property creates a new stacking context in the browser. Perhaps something you were doing was also modifying the stacking context? I wasn't able to find a direct explanation to my issue but perhaps somebody else would be able to break it down more.
